ansible/ocs.yaml

```diff
@@ -0,0 +1,292 @@
+---
+- hosts: convergence_base
+  gather_facts: false
+  become: true
+  user: root
+  vars_files: vars/default.yaml
+  roles:
+  - oc_local
+
+  tasks:
+  - name: Provision storage nodes and install OCS
+    when: ocs_enabled == true
+    block:
+    - name: Install non-pip dependencies
+      package:
+        name: "{{ item }}"
+        state: latest
+      with_items:
+        - ipmitool
+        - virt-install
+
+    - name: Install pip dependencies
+      command: "pip3 install {{ item }}"
+      with_items:
+        - virtualbmc
+
+    - name: Clear host VM memory cache
+      shell: echo 3 | tee /proc/sys/vm/drop_caches
+
+    - name: Create storage VMs
+      shell: |
+        virt-install --ram {{ ocp_storage_memory }} --vcpus {{ ocp_storage_vcpu }} --os-variant rhel8.0 --cpu host-passthrough \
+        --disk size={{ ocp_storage_disk }},pool=default,device=disk,bus=virtio,format=qcow2 --import --noautoconsole \
+        --vnc --network=bridge:{{ ocp_cluster_name}}pr,mac="52:54:00:82:68:6{{ item }}" \
+        --network=bridge:{{ ocp_cluster_name }}bm,mac="52:54:00:82:69:6{{ item }}" --name "{{ ocp_cluster_name}}-storage-{{ item }}" \
+        --os-type=linux --events on_reboot=restart --boot hd,network --noreboot
+      register: create_vms
+      failed_when: create_vms.stderr != "" and "is in use by another virtual machine" not in create_vms.stderr
+      with_items:
+        - 0
+        - 1
+        - 2
+
+    - name: Read host {{ ocs_device }} information
+      parted: device={{ ocs_device }} unit=GiB
+      register: device_info
+
+    - debug:
+        var: device_info
+
+    - name: Prepare partitions on {{ ocs_device }}
+      when: (device_info.partitions | length) == 0
+      block:
+        - name: Set msdos disk extended partition range end
+          set_fact:
+            extended_part_end: "100%"
+          when: device_info.disk.table == "msdos"
+
+        - name: Set non-msdos disk extended partition range end
+          set_fact:
+            extended_part_end: "40%"
+          when: device_info.disk.table != "msdos"
+
+        - name: Create partitions on {{ ocs_device }}
+          shell: |
+            parted {{ ocs_device }} mkpart primary ext2 0% 10%;
+            parted {{ ocs_device }} mkpart primary ext2 10% 20%;
+            parted {{ ocs_device }} mkpart primary ext2 20% 30%;
+            parted {{ ocs_device }} mkpart extended 30% {{ extended_part_end }};
+            parted {{ ocs_device }} mkpart logical ext2 40% 50%;
+            parted {{ ocs_device }} mkpart logical ext2 50% 60%;
+            parted {{ ocs_device }} mkpart logical ext2 60% 70%;
+            parted {{ ocs_device }} mkpart logical ext2 70% 80%;
+            parted {{ ocs_device }} mkpart logical ext2 80% 90%;
+            parted {{ ocs_device }} mkpart logical ext2 90% 100%
```

Comment:
Why not just use sparse files? Much simpler to manage and we don't risk blowing away data on an unusually-configured host.

Reply:
Right on.  Changed.